src/Containers.hs

```diff
@@ -479,19 +481,19 @@ asum = foldr (<|>) empty
 -- Disallowed instances
 ----------------------------------------------------------------------------
 
-#define DISALLOW_CONTAINER_8(t, z) \
+#define DISALLOW_TO_LIST_8(t, z) \
     instance TypeError \
         (Text "Do not use 'Foldable' methods on " :<>: Text z :$$: \
          Text "NB. If you tried to use 'for_' on Maybe or Either, use 'whenJust' or 'whenRight' instead" ) => \
```

Comment:
Currently we have have a lot of functions which can help you in absence of `Foldable` type class:

`whenJust` `whenRight` instead of `for_`
`maybeToMonoid` instead of `fold`

Could you make this helpful message more helpful and beautiful?

Like, so user can see something like this:

```
Do not use 'Foldable' methods on Maybe
Replacements:
    Instead of 
        for_ :: for_ :: (Foldable t, Applicative f) => t a -> (a -> f b) -> f ()
    use
        whenJust  :: Applicative f => Maybe a    -> (a -> f ()) -> f ()
        whenRight :: Applicative f => Either l r -> (r -> f ()) -> f ()

    Instead of
        fold :: (Foldable t, Monoid m) => t m -> m
    use 
        maybeToMonoid :: Monoid m => Maybe m -> m
```

Because you will reuse this error for both `DISALLOW_TO_LIST` and `DISALLOW_CONTAINER` you can create separate value using this module:

https://www.stackage.org/haddock/lts-9.14/base-4.9.1.0/GHC-TypeLits.html#v::-60--62-: